### PR TITLE
chore: unify schemes and test plans - WPB-11406

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11406" title="WPB-11406" target="_blank">StoryWPB-11406</a>  Refactor CI tests workflow
  </td></table>

### Issue

As a preparation for PR https://github.com/wireapp/wire-ios/pull/1707 this PR

- All projects get an "AllTests" tesplan
- TestPlans can now be reached in the Project Navigator
- Unused schemes are deleted
- Some Schemes were recreated, removing legacy content like skipped tests, run arguments etc.
- Added IDEWorkspaceSharedSettings_AutocreateContextsIfNeeded to the workspace configuration

Screenshot 2024-11-28 at 09 30 02

### Testing

Automatic tests should pass.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: .github/workflows/_reusable_run_tests.yml [hunk 1]
```diff
@@ -469,35 +469,17 @@ jobs:
         if: ${{ inputs.wire-api || inputs.all }}
         run: |
           echo "Testing WireAPI..."
-          xcodebuild test -workspace wire-ios-mono.xcworkspace -scheme WireAPI -destination 'platform=iOS Simulator,OS=${{ env.IOS_VERSION }},name=${{ env.IPHONE_MODEL }}' | tee -a xcodebuild-wire-api.log | bundle exec xcpretty --report junit --output build/reports/WireAPI.junit
+          xcodebuild test -workspace wire-ios-mono.xcworkspace -scheme WireAPI-Package -destination 'platform=iOS Simulator,OS=${{ env.IOS_VERSION }},name=${{ env.IPHONE_MODEL }}' | tee -a xcodebuild-wire-api.log | bundle exec xcpretty --report junit --output build/reports/WireAPI.junit
           exit ${PIPESTATUS[0]}
 
-      # WireDatadog
```
Comment (caldrian) [inline .github/workflows/_reusable_run_tests.yml:475]: DataDog will be compiled within the target WireAnalytics-Package.
There are no additional tests for WireDatadog.

File: wire-ios-link-preview/WireLinkPreviewTests/TestPlans/E2ETests.xctestplan [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  ],
+  "defaultOptions" : {
+    "language" : "en",
+    "region" : "DE",
+    "testExecutionOrdering" : "random"
+  },
+  "testTargets" : [
+    {
+      "skippedTests" : [
+        "ImageDownloaderTests",
```
Comment (caldrian) [inline wire-ios-link-preview/WireLinkPreviewTests/TestPlans/E2ETests.xctestplan:19]: This is a copy of the `AllTests` but all tests being disabled and only E2ETests enabled.

File: wire-ios-request-strategy/Tests/TestPlans/AllTests.xctestplan [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        "value" : "1"
+      }
+    ],
+    "language" : "en",
+    "region" : "DE",
+    "testExecutionOrdering" : "random"
+  },
+  "testTargets" : [
+    {
+      "skippedTests" : [
```
Comment (caldrian) [inline wire-ios-request-strategy/Tests/TestPlans/AllTests.xctestplan:30]: Setting these here, removing it from the scheme `wire-ios-request-strategy/WireRequestStrategy.xcodeproj/xcshareddata/xcschemes/WireRequestStrategy.xcscheme`

**Review by netbe — APPROVED** [review]

**Review by samwyndham — COMMENTED** [review]
Hey @caldrian. Two questions:

1. Could you give a bit more information in the description about what will specifically be different once this PR merged. Just because there is a lot of changes.
2. Just to check should this PR definitely be targeting release brach and not develop.

**Review by samwyndham — APPROVED** [review]

**Pull request merged**